### PR TITLE
redux

### PR DESCRIPTION
**Pull request by Anastasiya-Shir**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
 import NotFoundPage from './components/NotFoundPage';
-import Description from './components/Disription';
+import Description from './components/Description/Disription';
+
 import { BrowserRouter as Router, Routes, Route, } from 'react-router-dom';
-import BasicModal from './components/Modal';
+import BasicModal from './components/Modal/Modal';
 function App() {
+
   const [open, setOpen] = useState(false);
```
Comment (kathrynowy) [inline src/App.js]: isOpen - bool variable

File: src/components/Header/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import { getFilm } from '../store/searchSlice';
+
+function Header(props) {
+  const [isUserAuthorized, setIsUserAuthorized] = useState("");
+
+  const [inputSearch, setInputSearch] = useState('');
+
+  const debouncedSearchTerm = useDebounce(inputSearch, 1000);
+
+  const { setOpen, items, setFindFilms } = props;
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx]: move destructuring to the top

File: src/components/Header/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import { useSelector, useDispatch } from 'react-redux';
+
+import logo from '../../logo.png';
+
+import useDebounce from "../../service/use-debounce";
+
+import { getFilm } from '../store/searchSlice';
+
+function Header(props) {
+  const [isUserAuthorized, setIsUserAuthorized] = useState("");
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx]: use only '' , not ""

File: src/components/Header/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+  const { setOpen, items, setFindFilms } = props;
+
+  const searchFilms = useSelector(state => state.search.search);
+
+  const dispatch = useDispatch();
+
+
+  window.addEventListener('storage', (e) => {
+    const emailJson = localStorage.getItem('isUserAuthrized');
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx]: remove logic with local storage

File: src/components/Header/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    const usersEmail = JSON.parse(emailJson) || [];
+
+    setIsUserAuthorized(usersEmail.email);
+
+    setOpen(false);
+  })
+
+  let searchFilm = () => {
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx:31]: useCallback and const

File: src/components/Header/Header.jsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    let findFilms = items.filter(items =>
+      items.acronym.toUpperCase().includes(searchFilms[searchFilms.length - 1].searchFilms.toUpperCase()))
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx:34]: move this to useSelector

File: src/components/Header/Header.jsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      setFindFilms(findFilms)
+    }
+  };
+
+  useEffect(() => {
+    const newMovies = inputSearch;
+
+    setInputSearch(newMovies);
+
+    dispatch(getFilm({ inputSearch }));
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx:52]: why do you pass an object here?

File: src/components/Header/Header.jsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+    const usersauth = JSON.parse(auth) || [];
+
+    setIsUserAuthorized(usersauth.email);
+  }, []);
+
+  return (
+    <div className='header'>
+
+      <form onSubmit={e => e.preventDefault()}>
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx]: remove inline functions

File: src/components/Header/Header.jsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+        <input type="search" onChange={function (e) {
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx]: same here

File: src/components/Header/Header.jsx [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+          setInputSearch(e.target.value)
+        }
+        } placeholder='serch'>
+
+        </input>
+      </form>
+
+      <img src={logo} className="logo" alt=" movie poster" />
+      {isUserAuthorized === undefined
+        ? <button type="button" onClick={function () { setOpen(true) }}> Sign in  </button>
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx]: remove inline function everywhere and write me in TG why is it bad to write inline funcs

File: src/components/Movies/Movies.jsx [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+  const { items, setItems, findFilms } = props;
+
+  const dispatch = useDispatch();
+
+  const status = useSelector(state => state.AddMovies.status);
+
+  const error = useSelector(state => state.AddMovies.error);
+
+  const item = useSelector(state => state.AddMovies.films)
```
Comment (kathrynowy) [inline src/components/Movies/Movies.jsx]: write useSelectors properly
and we don't name field with a capital letter AddMovies - addMovies

File: src/components/Movies/Movies.jsx [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+  function changeSelect(event) {
+    if (event.target.value === "Минск") {
+      setValue(1);
+
+      return setValue(1);
```
Comment (kathrynowy) [inline src/components/Movies/Movies.jsx]: ?

File: src/components/Movies/Movies.jsx [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    } else {
+
+      return (setValue(5))
```
Comment (kathrynowy) [inline src/components/Movies/Movies.jsx]: you don't need to write return here

File: src/components/Movies/Movies.jsx [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+    }
+  };
+
+
+  useEffect(() => {
+    dispatch(fetchMovies());
+  }, [value]);
```
Comment (kathrynowy) [inline src/components/Movies/Movies.jsx:40]: why do you need value here?

File: src/components/Movies/Movies.jsx [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+  if (!status) {
+
```
Comment (kathrynowy) [inline src/components/Movies/Movies.jsx]: no need to paste an empty line

File: src/components/Movies/Movies.jsx [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+        ?
+        (<div> “Ooops, something went wrong” </div>)
+        : (
+          !item ?
+            (<p>We’ve found no movies, sorry!</p>)
+            :
+            (<>
+              <select onChange={changeSelect} className="selectCity">
+                <option>Минск</option>
+                <option>Гродно</option>
```
Comment (kathrynowy) [inline src/components/Movies/Movies.jsx]: remove ternaries and move above with if else

File: src/components/Movies/Movies.jsx [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+              </select>
+              {findFilms.length > 0 ? findFilms.map((item) => <Movie key={item.eventId} item={item} />) : item.map((item) => <Movie key={item.eventId} item={item} />)}
```
Comment (kathrynowy) [inline src/components/Movies/Movies.jsx]: remove it

always use filteredMovies
if the search is empty - add all movies to filtered

Comment (kathrynowy) [inline src/components/Movies/Movies.jsx]: and we don't check for 0 like this

!findFilms.length

File: src/components/store/index.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+import searchReducer from "./searchSlice";
+import toggleIsFetchingReducer from './thunk/LoadingMovieList'
+// import toggleIsFetchingReducer from "./toggleIsFetchingReducer"
+export default configureStore({
+  reducer: {
+    search: searchReducer,
+    AddMovies: toggleIsFetchingReducer,
```
Comment (kathrynowy) [inline src/components/store/index.js]: be careful with naming

File: src/components/store/searchSlice/index.js [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+
+  name: "search",
+  initialState: {
+    search: [],
+  },
+  reducers: {
+    getFilm(state, action) {
+      state.search.push({
+        id: ++i,
+        searchFilms: action.payload.inputSearch,
```
Comment (kathrynowy) [inline src/components/store/searchSlice/index.js]: searchQuery

File: src/components/store/thunk/LoadingMovieList.jsx [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+
+
+// const getMoviesListThunk = (value) => {
+//   async (dispatch) => {
+//     const data = await axios.get(`https://soft.silverscreen.by:8443/wssite/webapi/event/data?filter=%7B%22city%22:${value}%7D&extended=true`)
+//     const item = await data.json();
+//     console.log(item)
+//     return item
+//   }
+// }
```
Comment (kathrynowy) [inline src/components/store/thunk/LoadingMovieList.jsx]: ?

File: src/components/store/thunk/LoadingMovieList.jsx [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+import { createAsyncThunk, createSlice, isRejectedWithValue } from "@reduxjs/toolkit";
+
+export const fetchMovies = createAsyncThunk(
+
+  'fetchFilms/fetchMovies',
+  async function (_, isRejectedWithValue) {
+
+    const value = 1;
+    try {
+      const data = await fetch(`https://soft.silverscreen.by:8443/wssite/webapi/event/data?filter=%7B%22city%22:${value}%7D&extended=true`)
```
Comment (kathrynowy) [inline src/components/store/thunk/LoadingMovieList.jsx]: use axios

File: src/components/store/thunk/LoadingMovieList.jsx [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+      return item;
+    } catch (error) {
+      return isRejectedWithValue(error.messege)
+    }
+
+    // const item = []; 
+
+
+
+  }
```
Comment (kathrynowy) [inline src/components/store/thunk/LoadingMovieList.jsx]: ?

Comment (kathrynowy) [reply src/components/store/thunk/LoadingMovieList.jsx]: add eslint

File: src/service/use-debounce.jsx [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+      // Вернуть функцию очистки, которая будет вызываться каждый раз, когда ...
+      // ... useEffect вызван снова. useEffect будет вызван снова, только если ...
+      // ... value будет изменено (смотри ниже массив зависимостей).
+      // Так мы избегаем изменений debouncedValue, если значение value ...
+      // ... поменялось в рамках интервала задержки.
+      // Таймаут очищается и стартует снова.
+      // Что бы сложить это воедино: если пользователь печатает что-то внутри ...
+      // ... нашего приложения в поле поиска, мы не хотим, чтобы debouncedValue...
+      // ... не менялось до тех пор, пока он не прекратит печатать дольше, чем 500ms.
+      return () => {
```
Comment (kathrynowy) [inline src/service/use-debounce.jsx]: ?

**Review by kathrynowy — CHANGES_REQUESTED** [review]

File: src/components/Header/Header.jsx (hunk 2 above)
Comment (Anastasiya-Shir) [reply src/components/Header/Header.jsx]: fixed

File: src/App.css [hunk 24]
```diff
@@ -15,17 +15,63 @@
   width: 80%;
 }
 
-.button {
+.header button {
+  width: 150px;
+  height: 40px;
+  border: none;
+  border-radius: 20px;
+  color: #7fffd4;
```
Comment (kathrynowy) [inline src/App.css:22]: move to variables sass

File: src/App.css [hunk 25]
```diff
@@ -15,17 +15,63 @@
...
+  border: none;
+  margin-left: 10px;
+}
+
+.header .logoSearch {
+  display: flex;
+  flex-direction: row;
+}
+
+.Main {
```
Comment (kathrynowy) [inline src/App.css]: we don't name classnames with the capital letter

File: src/App.css [hunk 26]
```diff
@@ -15,17 +15,63 @@
...
+  border: none;
+  background-color: #20B2AA;
+  margin: 10px center;
+  text-align: center;
+  border-radius: 10px;
+}
+
+.movieList {
+  display: flex;
+  flex-direction: row;
```
Comment (kathrynowy) [inline src/App.css]: it's a default value

File: src/App.css [hunk 27]
```diff
@@ -147,6 +227,4 @@
 
 .offLogin {
   display: none;
-}
-
-/*# sourceMappingURL=App.css.map */
+}/*# sourceMappingURL=App.css.map */
```
Comment (kathrynowy) [inline src/App.css]: ?

File: src/App.js [hunk 28]
```diff
@@ -1,27 +1,36 @@
...
-import Description from './components/Disription';
+import Description from './components/Description/Disription';
+
 import { BrowserRouter as Router, Routes, Route, } from 'react-router-dom';
-import BasicModal from './components/Modal';
+import BasicModal from './components/Modal/Modal';
+
 function App() {
+
   const [open, setOpen] = useState(false);
```
Comment (kathrynowy) [inline src/App.js]: isOpen

File: src/App.js [hunk 29]
```diff
@@ -1,27 +1,36 @@
...
 import './App.css';
-import Header from './components/Header';
-import Movies from './components/Movies';
+import Header from './components/Header/Header';
+import Movies from './components/Movies/Movies';
 import NotFoundPage from './components/NotFoundPage';
-import Description from './components/Disription';
+import Description from './components/Description/Disription';
+
 import { BrowserRouter as Router, Routes, Route, } from 'react-router-dom';
```
Comment (kathrynowy) [inline src/App.js:8]: change order of imports pls
as i mentioned on a call

File: src/App.scss [hunk 30]
```diff
@@ -8,31 +8,80 @@ $aquamarine:#7fffd4;
...
+input{
+  border-radius:10px ;
+  width: 150px;
+  height: 30px;
+  padding:10px;
+  border:none;
+  margin-left: 10px;
+  
 }
+.logoSearch{
```
Comment (kathrynowy) [inline src/App.scss:52]: class-name-we-write-like-this
don't forget pls

File: src/components/AlertSigIn/AlertSigIn.jsx [hunk 31]
```diff
@@ -0,0 +1,36 @@
...
+import { useEffect } from 'react';
+
+import { Form, Field } from 'react-final-form'
+
+
+import Styles from "../ModalSigin/Styles";
+import { useSelector, useDispatch } from 'react-redux';
+import { isModalOpen } from '../store/ModalSlice';
+import Button from '@mui/material/Button';
+
```
Comment (kathrynowy) [inline src/components/AlertSigIn/AlertSigIn.jsx:11]: .

File: src/components/AlertSigIn/AlertSigIn.jsx [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+
+export default function AlertSegnIn() {
+
```
Comment (kathrynowy) [inline src/components/AlertSigIn/AlertSigIn.jsx]: 1 определись как ты делаешь экспорты компонентов, везде по-разному
2 ненужная пустая строка поле открытия компонента/функции И так далее

File: src/components/Header/Header.jsx [hunk 33]
```diff
@@ -0,0 +1,99 @@
...
+
+  const [inputSearch, setInputSearch] = useState('');
+
+  const debouncedSearchTerm = useDebounce(inputSearch, 1000);
+
+  const searchFilms = useSelector(state => state.search.search);
+
+  const authorized = useSelector(state => state.authorized.email);
+
+  const dispatch = useDispatch();
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx:27]: вынести в стор в файл selectors как я показывала

File: src/components/Header/Header.jsx [hunk 34]
```diff
@@ -0,0 +1,99 @@
...
+
+    let findFilms = items.filter(items =>
+      items.acronym.toUpperCase().includes(searchFilms[searchFilms.length - 1].searchFilms.toUpperCase()))
+
+    if (findFilms) {
+      setFindFilms(findFilms)
+    }
+  };
+
+  function Def(e) {
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx]: def?

File: src/components/Header/Header.jsx [hunk 35]
```diff
@@ -0,0 +1,99 @@
...
+    e.preventDefault();
+  }
+
+  function search(e) {
+    setInputSearch(e.target.value)
+  }
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx]: почему не стрелочная?
название handleSearch

File: src/components/Header/Header.jsx [hunk 36]
```diff
@@ -0,0 +1,99 @@
...
+
+  useEffect(() => {
+    const newMovies = inputSearch;
+
+    setInputSearch(newMovies);
+
+    dispatch(getFilm({ inputSearch }));
```
Comment (kathrynowy) [inline src/components/Header/Header.jsx:52]: зачем переменная setInputSearch

**Review by kathrynowy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**